packages/edit-site/src/components/page-templates/template-actions.js

```diff
@@ -0,0 +1,228 @@
+/**
+ * WordPress dependencies
+ */
+import { backup, trash } from '@wordpress/icons';
+import { __, sprintf } from '@wordpress/i18n';
+import { useDispatch } from '@wordpress/data';
+import { useMemo, useState } from '@wordpress/element';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+import { decodeEntities } from '@wordpress/html-entities';
+import {
+	Button,
+	TextControl,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+import isTemplateRevertable from '../../utils/is-template-revertable';
+import isTemplateRemovable from '../../utils/is-template-removable';
+import { TEMPLATE_POST_TYPE } from '../../utils/constants';
+
+export function useResetTemplateAction() {
```

Comment:
I presume this could also use `isDestructive`, given that it removes the user edits, bringing back the template to its original state.

Reply:
Currently we don't have it as `destructive` in templates list, so we can keep it as is for now. Small change to make if we need to.